chidioguejiofor/airtech-api
# #165062689: User should book a flight

#### What does this PR achieve?
- initialises the repo
- converts all db_column names to snake_case
- refactored the tests to use mocks
- moved conftest to tests directory
- adds booking app to the project
- adds tests to the app
- ensures that all tests added are passing



#### Any background context?
- Adds models, migrations and POST endpoint for the `/api/v1/flight/<flight-id>/booking` resource

#### New environmental Variables?
- None

#### Manual Testing Steps
- Migrate your data to your database via `python manage.py makemigrations && python manage.py migrate`
- Create several flights in the app
- Hit the POST `/api/v1/flight/<flight-id>/booking`  endpoint with the following request header and body:

*Header*
```
{ 'Authorization`: `Bearer <token>'}
```

You should get  this response


```
{
    "status": "success",
    "message": "You have successfully booked this Flight",
    "data": {
        "id": "c4510c96-1341-4c63-9d3b-42b172a2494c",
        "ticketPrice": "4000.70",
        "flight": {
            "id": "6b430b34-01f4-4aa3-a155-1241d357aa97",
            "capacity": 70,
            "location": "Popo York",
            "destination": "Lagos, Nigeria",
            "schedule": "2019-10-10T14:00:01Z",
            "currentPrice": "4000.70",
            "type": "international",
            "createdAt": "2019-04-08T15:42:27.402903Z",
            "updatedAt": null
        },
        "bookedBy": {
            "id": "8b963656-295d-48b0-92c6-6d0876ad1904",
            "firstName": "Prof",
            "gender": "Male",
            "lastName": "fdajhfio",
            "email": "prof1@email11.com",
            "username": "Prof10111"
        },
        "createdAt": "2019-04-10T04:05:51.457807Z",
        "updatedAt": null,
        "expiryDate": "2019-09-20T14:00:01Z"
    }
}
```
